pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy

```diff
@@ -122,6 +117,63 @@ class ModelInterpreter implements Serializable {
         }
     }
 
+    /**
+     * TODO
+     */
+    void runDirectiveWithPostBuild(Root root, Throwable firstErrorSeen, Closure body) {
+        try {
+            body.call()
+        } catch (Throwable e) {
+            if (firstErrorSeen == null) {
+                firstErrorSeen = e
+            }
+            // TODO: Ignoring errors in post because we already errored in a directive anyway so this would just mask things.
```

Comment:
Would it be useful to track these exceptions as suppressed exceptions rather than just dropping them, or are exceptions after the first one generally useless to the end user? I am imagining something like:

    if (firstErrorSeen == null) {
        firstErrorSeen = e
    } else {
        firstErrorSeen.addSuppressed(e)
    }
    try { 
        executePostBuild(root)
    } catch (Throwable e2) {
        firstErrorSeen.addSuppressed(e2)
    }